spark/src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommandBase.scala

```diff
@@ -228,7 +228,9 @@ abstract class MergeIntoCommandBase extends LeafRunnableCommand
       txn: OptimisticTransaction,
       outputDF: DataFrame): Seq[FileAction] = {
     val partitionColumns = txn.metadata.partitionColumns
-    if (partitionColumns.nonEmpty && spark.conf.get(DeltaSQLConf.MERGE_REPARTITION_BEFORE_WRITE)) {
+    // If the write will be optimized write, which shuffles the data anyway, then don't repartition.
```

Comment:
can you mention in the comment that Optimized write will handle both cases of splitting the task when its very large and combining tasks when they are very small.

Reply:
Updated the comment.